src/electron/platform/android/scan-results-to-unified-results.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { InstanceResultStatus, UnifiedResult } from 'common/types/store-data/unified-data-interface';
+import { UUIDGeneratorType } from 'common/uid-generator';
+import { RuleInformation } from './rule-information';
+import { RuleInformationProvider } from './rule-information-provider';
+import { RuleResultsData, ScanResults } from './scan-results';
+
+export type ConvertScanResultsToUnifiedResultsDelegate = (scanResults: ScanResults, uuidGenerator: UUIDGeneratorType) => UnifiedResult[];
+
+export function convertScanResultsToUnifiedResults(scanResults: ScanResults, uuidGenerator: UUIDGeneratorType): UnifiedResult[] {
+    if (scanResults == null || scanResults.ruleResults == null) {
+        return [];
+    }
+
+    return createUnifiedResultsFromScanResults(scanResults.ruleResults, uuidGenerator);
+}
+
+function createUnifiedResultsFromScanResults(ruleResults: RuleResultsData[], uuidGenerator: UUIDGeneratorType): UnifiedResult[] {
+    const ruleInformationProvider: RuleInformationProvider = new RuleInformationProvider();
```

Comment:
In general, we don't instanciate dependencies outside the composition root (`src\electron\device-connect-view\device-connect-view-initializer.ts` in the case of AI-Android) or factory classes (which we don't have on AI-Android). This works good enough for classes, which is not the case here.
In the case of functions, we usually follow the pattern of a creator function, where the creator function allows to bind any dependency (`src\issue-filing\services\azure-boards\create-azure-boards-issue-filing-url.ts` has an example of that).

All that said... I'm wondering if it make sense to have more than one instance of `RuleInformationProvider` as the class seems to contain "static" information of the rules we run on an android scan.

I'm not 100% sure how to change this right now, or if it's worth to change right now, my main intention is to share our general approach on how we handle instantiation and dependencies.

Reply:
Karan and I had a similar conversation. Long-term, we definitely only need one, but we were thinking of leaving it this way until we sort out the plumbing around calling the converter (since this will necessarily alter the signature of the converter delegates)